config/initializers/filter_parameter_logging.rb

```diff
@@ -4,37 +4,61 @@
 
 # Configure sensitive parameters which will be filtered from the log file.
 ALLOWLIST = %w[
-  controller
   action
-  id
-  from_date
-  to_date
-  qqtotalfilesize
-  type
-  folder_id
-  startDate
-  endDate
-  included
-  page
-  useCache
-  number
-  size
-  sort
-  showCompleted
-  excludeProvidedMessage
-  document_id
-  document_type
   category
-  cookie_id
-  reply_id
-  ids
   code
+  controller
+  cookie_id
+  document_id
+  document_type
+  endDate
   endpoint_sid
+  excludeProvidedMessage
+  filter
+  folder_id
+  from_date
+  id
+  ids
+  included
   message_id
+  number
   os_name
-  filter
+  page
+  qqtotalfilesize
+  reply_id
+  showCompleted
+  size
+  sort
+  startDate
   startedFormVersion
+  to_date
+  type
+  useCache
 ].freeze
-Rails.application.config.filter_parameters = [lambda do |k, v|
-  v.replace('FILTERED') if v.is_a?(String) && ALLOWLIST.exclude?(k)
-end]
+
+Rails.application.config.filter_parameters = [
+  lambda do |key, value|
+    # If the parameter key is not allowed, filter the value.
+    if ALLOWLIST.include?(key)
+      # Key is in allowlist, so leave value as is.
+      value
+    else
+      case value
+      when String
+        # For strings, we can mutate in place.
+        value.replace 'FILTERED'
+      when Numeric
+        # Numbers are immutable; return a filtered string.
+        'FILTERED'
+      when ActionDispatch::Http::UploadedFile
+        # For uploaded files, filter out the filename.
+        'FILTERED FILE'
```

Comment:
This doesn't seem to actually filter the file. I also tried setting `value = 'FILTERED FILE'` and that didn't work either.